frontend/src/libs/compare-table-rows.ts

```diff
@@ -0,0 +1,33 @@
+import { Assignment } from "../api/defs/types";
+import { ConfirmationDdahRowData } from "../views/admin/ddahs/manipulate-ddah-confirmation";
+
+function compareString(str1: string, str2: string) {
+    if (str1 > str2) {
+        return 1;
+    } else if (str1 < str2) {
+        return -1;
+    }
+    return 0;
+}
+
+export function compareAssignment(a1: Assignment, a2: Assignment) {
+    return (
+        compareString(a1.position.position_code, a2.position.position_code) ||
+        compareString(
+            a1.applicant.last_name || "",
+            a2.applicant.last_name || ""
+        ) ||
+        compareString(a1.applicant.first_name, a2.applicant.first_name)
+    );
+}
+
+export function compareDDAH(
+    d1: ConfirmationDdahRowData,
```

Comment:
Should be a `Pick<Ddah, "position_code" | "last_name" | "first_name">`. You shouldn't need to import from a view into a utility. If you've doe such a thing, it's an indication that the code isn't organized correctly.

Reply:
The `DDAH`s have `position_code`, `last_name`, and `first_name` nested in `assignment.position` or `assignment.applicant`. Using the `Pick` would require some implementation or libraries to support nested `Pick`. which I dont think it's much easier to understand than the original interface `ConfirmationDdahRowData`. Are we going to use Pick in this case?

Reply:
Another easier solution I think I can go is to modify the compare DDAH method itself

Reply:
Then just define a new type `{position_code: string, last_name: string, first_name: string}`